Backpack/src/main/java/net/skyscanner/backpack/util/ItemHolder.kt

```diff
@@ -0,0 +1,39 @@
+package net.skyscanner.backpack.util
+
+import android.content.Context
+import android.content.res.Resources
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.IdRes
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.RecyclerView
+
+internal abstract class ItemHolder<T>(
+  protected val view: View
+) : RecyclerView.ViewHolder(view), Consumer<T> {
+
+  val context: Context
+    get() = itemView.context
+
+  val resources: Resources
+    get() = itemView.resources
+
+  var model: T? = null
+    private set
+
+  constructor(parent: ViewGroup, @LayoutRes layout: Int) :
```

Comment:
nit: constructor first

Reply:
According to the Kotlin guidelines, the properties should be declared first:

https://kotlinlang.org/docs/reference/coding-conventions.html#class-layout

Reply:
That's new to me. I find it a bit confusing that they say to keep related stuff together and also recommend not to keep constructors together by declaring the props first 🤷. But not a blocker was just a nit pick